### DISCUSSION TITLE
KNEM and CMA  bandwidth

### DISCUSSION THREAD

**Discussion by w7658610** [openucx/ucx#8199] [Q&A] [answered]
knem:
> bandwidth: 13862.00/ppn + 0.00 MB/sec

cma:
> bandwidth: 0.00/ppn + 11145.00 MB/sec

In UCX, CMA bandwidth is dedicated（bandwidth.shared = 0）, but KNEM bandwidth is shared（bandwidth.dedicated = 0）,why？

![image](https://user-images.githubusercontent.com/37858414/167134805-47fcd17c-f199-43dc-8798-04bc20a80999.png)

![image](https://user-images.githubusercontent.com/37858414/167134880-d4e79f49-d408-40df-9c57-247844c94e1b.png)

**Comment by dmitrygx** [comment, marked as answer]
@w7658610 KNEM uses a single thread in a kernel implementation and this thread is shared between all process. So, this thread becomes a bottleneck when increasing scale.
In the contrast, CMA doesn't have such weakness. So, UCP selects CMA instead of KNEM at some scale.
